src/_pytest/python_api.py

```diff
@@ -566,7 +565,6 @@ def raises(  # noqa: F811
 def raises(  # noqa: F811
     expected_exception: Union["Type[_E]", Tuple["Type[_E]", ...]],
     *args: Any,
-    match: Optional[Union[str, "Pattern"]] = None,
     **kwargs: Any
 ) -> Union["RaisesContext[_E]", Optional[_pytest._code.ExceptionInfo[_E]]]:
```

Comment:
@bluetech do the typing changes make sense?

Reply:
Yes, the types look good to me.

Reply:
is there a way to type it so, context manager usage and function usage get the different parameter sets correct?

so no args, but matches typed for contextmanger usage, and args, kwargs when the first "arg" is a callable?

Reply:
@RonnyPfannschmidt, I believe it already does that (unless I'm confused).

For callers, only the `@overloads` are used for type checking, and there are two overloads above, one for the context manager case and one for the function case.

The annotations on the function def without the `@overload` is just used for type checking the *implementation*. The implementation needs to handle both cases so the types use `Union` etc.